Classes/Util.php

```diff
@@ -274,22 +274,23 @@ public static function getConfigurationFromPageId(
 
         if ($initializeTsfe) {
             self::initializeTsfe($pageId, $language);
-            if (!empty($configurationToUse)) {
-                return self::buildTypoScriptConfigurationFromArray($configurationToUse, $pageId, $language, $path);
+            if (empty($configurationToUse)) {
+                $configurationToUse = self::getConfigurationFromInitializedTSFE($path);
             }
-            $configurationToUse = self::getConfigurationFromInitializedTSFE($path);
         } else {
-            if (!empty($configurationToUse)) {
-                return self::buildTypoScriptConfigurationFromArray($configurationToUse, $pageId, $language, $path);
+            if (empty($configurationToUse)) {
+                $configurationToUse = self::getConfigurationFromExistingTSFE($pageId, $path, $language);
```

Comment:
Using this function could break since TSFE is not initialised. I think this could be a mistake - that we want to use `buildTypoScriptConfigurationFromArray` instead of `getConfigurationFromExistingTSFE`. If there's a boost from using an existing TSFE and this code can potentially run in the frontend and/or works with a pre-initialised TSFE then we should check if one exists before using this method, and fallback to `buildTypoScriptConfigurationFromArray` if TSFE neither exists nor $initializeTsfe were true.